src/Sylius/Behat/Service/NotificationChecker.php

```diff
@@ -35,12 +36,25 @@ public function checkNotification(string $message, NotificationType $type): void
         foreach ($this->notificationAccessor->getMessageElements() as $messageElement) {
             if (
                 false !== strpos($messageElement->getText(), $message) &&
-                $messageElement->hasClass($type->__toString() === 'success' ? 'positive' : 'negative')
+                $messageElement->hasClass($this->resolveClass($type))
             ) {
                 return;
             }
         }
 
         throw new NotificationExpectationMismatchException($type, $message);
     }
+
+    private function resolveClass(NotificationType $type): string
+    {
+        $typeClassMap = [
+            'failure' => 'negative',
+            'info' => 'info',
+            'success' => 'positive',
+        ];
+
+        Assert::keyExists($typeClassMap, $type->__toString());
```

Comment:
I'm just wondering, what is better
```suggestion
        Assert::keyExists($typeClassMap, (string) $type);
```